From 5985c61e19256d51f281d87e6610760663854b30 Mon Sep 17 00:00:00 2001
From: Kirill Smelkov <kirr@mns.spb.ru>
Date: Wed, 24 Nov 2010 21:27:58 +0300
Subject: [PATCH 3/4] Tech szBasename about '/' and '\\' path separators (win32 only)

Previously it was using '/' on linux and '\' on mingw. Since on windows
path separator can be on of those two, teach szBasename about it.

Signed-off-by: Kirill Smelkov <kirr@mns.spb.ru>
---
 misc.c |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/misc.c b/misc.c
index 609a2f0..63d1c91 100644
--- a/misc.c
+++ b/misc.c
@@ -519,7 +519,7 @@ unilen(const UCHAR *s)
 const char *
 szBasename(const char *szFilename)
 {
-	const char	*szTmp;
+	const char	*tail;
 
 	fail(szFilename == NULL);
 
@@ -527,11 +527,12 @@ szBasename(const char *szFilename)
 		return "null";
 	}
 
-	szTmp = strrchr(szFilename, FILE_SEPARATOR[0]);
-	if (szTmp == NULL) {
-		return szFilename;
+	/* check for both '/' and '\\' */
+	for (tail = szFilename+strlen(szFilename)-1; tail >= szFilename; --tail) {
+		if (*tail == '/' || *tail == '\\')
+			return tail+1;
 	}
-	return ++szTmp;
+	return szFilename;
 } /* end of szBasename */
 
 /*
-- 
1.7.3.2.245.g03276

